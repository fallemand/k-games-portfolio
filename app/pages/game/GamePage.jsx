import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import './game.scss';
import portfolioService from '../../services/portfolio.service';
import gamesService from '../../services/games.service';

class GamePage extends React.Component {
  constructor(props) {
    super(props);
    const { match } = props;
    const { id } = match.params;
    this.game = gamesService.getGame(id);
  }

  render() {
    const { game } = this;
    return (
      <div className="game">
        <div className="game__header">
          <h1 className="game__title">{game.name}</h1>
        </div>
        <div className="game__play">
          <img className="game__play-background" src={game.images.screenshot} alt={game.name} />
          <div className="game__play-overlay" />
          <div className="game__play-icon">
            Lets Play
          </div>
        </div>
      </div>
    );
  }
}

GamePage.propTypes = {
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
};

export default withRouter(GamePage);
