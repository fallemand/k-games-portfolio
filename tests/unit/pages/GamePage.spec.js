import React from 'react';
import GamePage from '../../../app/pages/game/GamePage';

jest.mock('../../../app/services/games.service', () => ({
  getGame: () => ({
    name: '__NAME__',
    short: '__ID__',
    images: {
      screenshot: '__SCREENSHOT-URL__',
    },
  }),
}));

jest.mock('../../../app/services/portfolio.service', () => ({
  contains: () => true,
}));


describe('GamePage', () => {
  const props = {
    history: { goBack: jest.fn() },
    match: { params: { id: 1 } },
  };

  it('render with props', () => {
    const component = shallow(<GamePage {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('change state and render', () => {
    const component = shallow(<GamePage {...props} />);
    component.setState({ inPortfolio: false });
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
