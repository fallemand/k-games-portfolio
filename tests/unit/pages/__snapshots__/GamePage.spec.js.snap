// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePage change state and render 1`] = `
"<div className='game'>
  <div className='game__header'>
    <h1 className='game__title'>
      __NAME__
    </h1>
    <Button className='game__button' onClick={[Function: onClick]} color='green'>
      Add to Portfolio
    </Button>
  </div>
  <div className='game__play'>
    <img className='game__play-background' src='__SCREENSHOT-URL__' alt='__NAME__' />
    <div className='game__play-overlay' />
    <div className='game__play-icon'>
      Lets Play
    </div>
  </div>
  <Snackbar message='' show={false} className={{...}} />
</div>"
`;

exports[`GamePage render with props 1`] = `
"<div className='game'>
  <div className='game__header'>
    <h1 className='game__title'>
      __NAME__
    </h1>
    <Button className='game__button' onClick={[Function: onClick]} color='default'>
      Remove from Portfolio
    </Button>
  </div>
  <div className='game__play'>
    <img className='game__play-background' src='__SCREENSHOT-URL__' alt='__NAME__' />
    <div className='game__play-overlay' />
    <div className='game__play-icon'>
      Lets Play
    </div>
  </div>
  <Snackbar message='' show={false} className={{...}} />
</div>"
`;
